src/plugins.star

```diff
@@ -0,0 +1,78 @@
+    "AutoGPT_IFTTT": {"repository": "AntonioCiolino/AutoGPT-IFTTT", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["IFTTT_WEBHOOK_TRIGGER_NAME", "IFTTT_KEY"]},
+    "AutoGPT_YouTube": {"repository": "jpetzke/AutoGPT-YouTube", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["YOUTUBE_API_KEY"]},
+    "AutoGPTPMPlugin": {"repository": "minfenglu/AutoGPT-PM-Plugin", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["TRELLO_API_KEY", "TRELLO_API_TOKEN", "TRELLO_CONFIG_FILE"]},
+    "AutoGPTWolframAlpha": {"repository":"gravelBridge/AutoGPT-WolframAlpha", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["WOLFRAM_ALPHA_APP_ID"]},
+    "AutoGPTTodoistPlugin": {"repository": "danikhan632/Auto-GPT-Todoist-Plugin", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["TODOIST_TOKEN"]},
+    "AutoGPTMessagesPlugin": {"repository": "danikhan632/Auto-GPT-Messages-Plugin", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["IMESSAGE_PASSWORD_KEY", "IMESSAGE_BASE_URL"]},
+    "AutoGPTWebInteraction": {"repository": "gravelBridge/AutoGPT-Web-Interaction", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: []},
+    "AutoGPTNotion": {"repository": "doutv/Auto-GPT-Notion", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["NOTION_TOKEN", "NOTION_DATABASE_ID"]},
+    "SystemInformationPlugin": {"repository": "hdkiller/Auto-GPT-SystemInfo", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: []},
+    "AutoGPT_Zapier": {"repository": "AntonioCiolino/AutoGPT-Zapier", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["ZAPIER_WEBHOOK_ENDPOINT"]},
+}
+
+def get_plugin_url(plugin_data, plugin_branch_to_use, plugin_repo_to_use):
+    repo = plugin_data["repository"]
+    if plugin_repo_to_use:
+        repo = plugin_repo_to_use
+    branch = plugin_data["branch"]
+    if plugin_branch_to_use:
+        branch = plugin_branch_to_use
+    return "https://github.com/{0}/archive/refs/heads/{1}.zip".format(repo, branch)
+
+def get_filename(plugin):
+    author, actual_repo = plugin["repository"].split("/")
+    return actual_repo + ZIP_EXTENSION
+
+def areAllRequiredEnvVarsSet(env_vars, plugins_names):
+    areAllRequiredEnvVarsSet = True
+    missing_env_vars = []
+    all_required_env_vars = getAllRequiredEnvVarsFor(plugins_names)
+    for required_env_var in all_required_env_vars:
+        if required_env_var not in env_vars:
+            areAllRequiredEnvVarsSet = False
+            missing_env_vars.append(required_env_var)
+
+    return areAllRequiredEnvVarsSet, missing_env_vars
+    
+
+def getAllRequiredEnvVarsFor(plugins_names):
+    required_env_vars = []
+    for plugin_name in plugins_names:
+        plugin_data = plugins_map.get(plugin_name)
+        if plugin_data == None:
+            fail("Invalid plugin name {0}. The supported plugins are: {1}. You can add support for a new plugin by creating an issue or PR at {2}".format(plugin_name, ", ".join(plugins_map.keys()), common.KURTOSIS_AUTOGPT_PACKAGE_URL))
```

Comment:
can we decouple this from the env var check!? so we can selectively disable the env var check